packages/rx-effects-react/src/useObserver.ts

```diff
@@ -12,32 +12,81 @@ import { useConst } from './useConst';
  *
  * @example
  * ```ts
- * const observer = useCallback((nextValue) => {
+ * useObserver(source$, (nextValue) => {
  *   logger.log(nextValue);
- * }, []);
- * useObserver(source$, observer);
+ * });
  * ```
  */
 export function useObserver<T>(
   source$: Observable<T>,
-  observerOrNext: Partial<Observer<T>> | ((value: T) => void),
+  observerOrNext: PartialObserver<T> | ((value: T) => void),
 ): Subscription {
   const hookSubscriptions = useConst(() => new Subscription());
 
+  const argsRef = useRef<PartialObserver<T> | ((value: T) => void)>(
+    observerOrNext,
+  );
+
+  // Update the latest observable and callbacks
+  // synchronously after render being committed
+  useIsomorphicLayoutEffect(() => {
+    argsRef.current = observerOrNext;
+  });
+
   useEffect(() => {
-    if (!hookSubscriptions.closed) {
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore: subscribe cannot infer the right type.
-      const subscription = source$.subscribe(observerOrNext);
-      hookSubscriptions.add(subscription);
-      return () => subscription.unsubscribe();
+    if (hookSubscriptions.closed) {
```

Comment:
I'm thinking about to remove `hookSubscriptions` at all (another PR).
This will be a breaking change I double that it is used by anyone.

Reply:
Approve. I don't understand when we need it